src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -327,7 +327,13 @@ private static Socket Connect(string serverName, int port, TimeSpan timeout, boo
             string IPv4String = null;
             string IPv6String = null;
 
-            Socket[] sockets = new Socket[2];
+            // Returning null socket is handled by the caller function.
+            if(ipAddresses == null || ipAddresses.Length == 0)
+            {
+                return null;
+            }
+
+            Socket[] sockets = new Socket[ipAddresses.Length];
```

Comment:
I quiet do not understand why we do not use `List` type instead of the array. In that case we won't need the extra incrementor later. Is there any problem using `List`?

Reply:
This is only as per current design, with changes coming from https://github.com/dotnet/SqlClient/pull/1029, this will change too.